sky/backends/backend_utils.py

```diff
@@ -481,226 +481,83 @@ def add_cluster(
-                    docker_proxy_command = docker_proxy_command_generator(ip)
-                    ip = 'localhost'
-                # TODO(romilb): Update port number when k8s supports multinode
-                codegens[idx] = cls._get_generated_config(
-                    sky_autogen_comment, host_name, ip, username, key_path,
-                    proxy_command, port, docker_proxy_command)
-
-        # This checks if all codegens have been created.
-        for idx, ip in enumerate(external_worker_ips):
-            if docker_user is not None:
-                docker_proxy_command = docker_proxy_command_generator(ip)
-                ip = 'localhost'
-            if not codegens[idx]:
-                codegens[idx] = cls._get_generated_config(
-                    sky_autogen_comment, worker_names[idx], ip, username,
-                    key_path, proxy_command, port, docker_proxy_command)
-
-        for idx in range(len(external_worker_ips)):
-            # Add (or overwrite) the new config.
-            overwrite = overwrites[idx]
-            overwrite_begin_idx = overwrite_begin_idxs[idx]
-            codegen = codegens[idx]
-            assert codegen is not None, (codegens, idx)
-            if overwrite:
-                assert overwrite_begin_idx is not None
-                updated_lines = codegen.splitlines(keepends=True) + ['\n']
-                extra_config[overwrite_begin_idx:overwrite_begin_idx +
-                             len(updated_lines)] = updated_lines
-                with open(extra_config_path, 'w') as f:
-                    f.write(''.join(extra_config).strip())
-                    f.write('\n' * 2)
-            else:
-                with open(extra_config_path, 'a') as f:
-                    f.write(codegen)
-                    f.write('\n')
+                ['-W', '%h:%p', f'{auth_config["ssh_user"]}@{ips[0]}'])
+            head_port = constants.DEFAULT_DOCKER_PORT
 
-        # Add trailing new line at the end of the file if it doesn't exit
-        with open(extra_config_path, 'r+') as f:
-            extra_config = f.readlines()
-            if extra_config[-1] != '\n':
-                f.write('\n')
+        # Start generating the config with the head node
+        codegen = cls._get_generated_config(
+            sky_autogen_comment, host_name, ip, username, key_path,
+            proxy_command, head_port, docker_proxy_command) + '\n'
+
+        # Add the worker nodes if any exist
+        for i, ip in enumerate(ips[1:]):
+            worker_name = cluster_name + f'-worker{i+1}'
+            # TODO: update port numbers for workers in edge cases
+            codegen += cls._get_generated_config(
+                sky_autogen_comment,
+                worker_name,
+                ip,
+                username,
+                key_path,
+                proxy_command,
+                ports[i + 1],  #updated port numbers
+                docker_proxy_command) + '\n'
```

Comment:
For docker support, the IP should be `localhost` instead and the `docker_proxy_command` should be filled with the worker IP so that the ssh can be proxied to the docker container.
https://github.com/skypilot-org/skypilot/blob/7609e229a448a8f7d27642a8d34d33b6e0534702/sky/backends/backend_utils.py#L640-L646